rest/endpoint/v1/routes.go

```diff
@@ -0,0 +1,175 @@
+package endpoint
+
+import (
+	"github.com/datastax/cassandra-data-apis/config"
+	"github.com/datastax/cassandra-data-apis/db"
+	"github.com/datastax/cassandra-data-apis/log"
+	"github.com/datastax/cassandra-data-apis/types"
+	"net/http"
+	"path"
+)
+
+const (
+	keyspaceParam = "keyspaceName"
+	tableParam    = "tableName"
+)
+
+const (
+	KeyspacesPathFormat    = "v1/keyspaces"
+	TablesPathFormat       = "v1/keyspaces/%s/tables"
+	TableSinglePathFormat  = "v1/keyspaces/%s/tables/%s"
+	ColumnsPathFormat      = "v1/keyspaces/%s/tables/%s/columns"
+	ColumnSinglePathFormat = "v1/keyspaces/%s/tables/%s/columns/%s"
+	RowsPathFormat         = "v1/keyspaces/%s/tables/%s/rows"
+	RowSinglePathFormat    = "v1/keyspaces/%s/tables/%s/rows/%s"
+	QueryPathFormat        = "v1/keyspaces/%s/tables/%s/rows/query"
+)
+
+// routeList describes how to route an endpoint
+type routeList struct {
+	logger            log.Logger
+	params            config.UrlParamGetter
+	dbClient          *db.Db
+	operations        config.SchemaOperations
+	excludedKeyspaces map[string]bool
+	singleKeyspace    string
+}
+
+// Routes returns a slice of all the REST endpoint routes
+func Routes(prefix string, operations config.SchemaOperations, singleKeyspace string, cfg config.Config, dbClient *db.Db) []types.Route {
+	excludedKeyspaces := make(map[string]bool)
+	for _, ks := range cfg.ExcludedKeyspaces() {
+		excludedKeyspaces[ks] = true
+	}
+
+	rl := routeList{
+		logger:            cfg.Logger(),
+		params:            cfg.RouterInfo().UrlParams(),
+		dbClient:          dbClient,
+		operations:        operations,
+		excludedKeyspaces: excludedKeyspaces,
+		singleKeyspace:    singleKeyspace,
+	}
+
+	urlPattern := cfg.RouterInfo().UrlPattern()
+
+	urlKeyspaces := url(prefix, urlPattern, KeyspacesPathFormat)
+	urlTables := url(prefix, urlPattern, TablesPathFormat, keyspaceParam)
+	urlSingleTable := url(prefix, urlPattern, TableSinglePathFormat, keyspaceParam, tableParam)
+	urlColumns := url(prefix, urlPattern, ColumnsPathFormat, keyspaceParam, tableParam)
+	urlSingleColumn := url(prefix, urlPattern, ColumnSinglePathFormat, keyspaceParam, tableParam, "columnName")
+	urlRows := url(prefix, urlPattern, RowsPathFormat, keyspaceParam, tableParam)
+	urlSingleRow := url(prefix, urlPattern, RowSinglePathFormat, keyspaceParam, tableParam, "rowIdentifier")
+	urlQuery := url(prefix, urlPattern, QueryPathFormat, keyspaceParam, tableParam)
+
+	routes := []types.Route{
+		{
+			Method:  http.MethodGet,
+			Pattern: urlColumns,
+			Handler: rl.validateKeyspace(rl.GetColumns),
+		},
+		{
+			Method:  http.MethodPost,
+			Pattern: urlColumns,
+			Handler: rl.validateKeyspace(rl.isSupported(config.TableAlterAdd, rl.AddColumn)),
+		},
+		{
+			Method:  http.MethodDelete,
+			Pattern: urlColumns,
```

Comment:
This should be `urlSingleColumn` for delete column.

Reply:
oh! good one! I wouldn't be able to catch that in a million years without testing it.